shutter.py

```diff
@@ -0,0 +1,196 @@
+CATEGORY_SHOWER_HEAD = 30
+'''
+
+logging.basicConfig(level=logging.INFO, format="[%(module)s] %(message)s")
+
+class TemperatureSensor(Accessory):
+    """Fake Temperature sensor, measuring every 3 seconds."""
+
+    category = CATEGORY_SENSOR
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        serv_temp = self.add_preload_service('TemperatureSensor')
+        self.char_temp = serv_temp.configure_char('CurrentTemperature')
+
+    @Accessory.run_at_interval(3)
+    async def run(self):
+        self.char_temp.set_value(random.randint(18, 26))
+
+
+class FakeFan(Accessory):
+    """Fake Fan, only logs whatever the client set."""
+
+    category = CATEGORY_FAN
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Add the fan service. Also add optional characteristics to it.
+        serv_fan = self.add_preload_service(
+            'Fan', chars=['RotationSpeed', 'RotationDirection'])
+
+        self.char_rotation_speed = serv_fan.configure_char(
+            'RotationSpeed', setter_callback=self.set_rotation_speed)
+        self.char_rotation_direction = serv_fan.configure_char(
+            'RotationDirection', setter_callback=self.set_rotation_direction)
+
+    def set_rotation_speed(self, value):
+        logging.debug("Rotation speed changed: %s", value)
+
+    def set_rotation_direction(self, value):
+        logging.debug("Rotation direction changed: %s", value)
+
+class LightBulb(Accessory):
+    """Fake lightbulb, logs what the client sets."""
+
+    category = CATEGORY_LIGHTBULB
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        serv_light = self.add_preload_service('Lightbulb')
+        self.char_on = serv_light.configure_char(
+            'On', setter_callback=self.set_bulb)
+
+    def set_bulb(self, value):
+        logging.info("Bulb value: %s", value)
+
+class GarageDoor(Accessory):
+    """Fake garage door."""
+
+    category = CATEGORY_GARAGE_DOOR_OPENER
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.add_preload_service('GarageDoorOpener')\
+            .configure_char(
+                'TargetDoorState', setter_callback=self.change_state)
+
+    def change_state(self, value):
+        logging.info("GDC value: %s", value)
+        self.get_service('GarageDoorOpener')\
+            .get_characteristic('CurrentDoorState')\
+            .set_value(value)
+
+
+class WindowCovering(Accessory):
+    """Fake WindowCovering."""
+
+    category = CATEGORY_WINDOW_COVERING
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        """Quick assign basic accessory information."""
+        self.set_info_service(firmware_revision=2,  manufacturer="Brand",
+                model="Shutter", serial_number="0123456789")
+        
+        # Add the WindowCovering service. Also add optional characteristics to it.
```

Comment:
This comment and the one about assigning accessory infos can be removed as well.